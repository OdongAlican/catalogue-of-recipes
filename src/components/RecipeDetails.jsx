import React from 'react'
import { Link } from 'react-router-dom'


const RecipeDetails = (props) => {

    const recipeData = props.location.state.recipe
    const dateArray = recipeData.publishedDate.split(' ')
    const dateValue = new Date(dateArray[0])
return (
    <div data-testid="recipeDetails">
        <div className="card p-2 col-6 border mx-auto mt-3 border" >
        <div  className="image-section-details mx-auto mb-2 border p-2">
            <img src={recipeData.image} alt="boohoo" className="image-detail"/>
        </div>
            <div className="card-body card-section">
                <h5 className="card-title text-underline"> {recipeData.title} </h5>
                <p className="card-text text-secondary">{ recipeData.text }</p>
                <p className="card-text text-underline"> Published Date: {
                dateValue.toDateString()} </p>
                <p className="card-text text-underline"> Web-site:                     
                <a href={recipeData.site} target = '_blank' className="ml-1" rel="noopener noreferrer"> {recipeData.site}</a>
                </p>
            </div>
            <button className="col-md-2 btn btn-primary">
                <Link to = {{ pathname: '/'}} className="text-white">
                     Home 
                </Link>
            </button>
        </div>
    </div>
)

}


export default RecipeDetails;