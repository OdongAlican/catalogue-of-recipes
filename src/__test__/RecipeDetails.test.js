import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { MemoryRouter, BrowserRouter as Router } from 'react-router-dom';
import RecipeDetails from '../components/RecipeDetails';

const location = {
  state: {
    recipe: {
      image: 'sample',
      title: 'title',
      site: 'site',
      publishedDate: '2020-09-23 15:00:00',
    },
  },
};

afterEach(cleanup);

it('renders recipeDetaills correctly', () => {
  const { getAllByTestId } = render(
    <RecipeDetails location={location} />, { wrapper: MemoryRouter },
  );
  const result = getAllByTestId('recipeDetails');
  expect(result[0]).toHaveTextContent('title Published Date: Wed Sep 23 2020 Web-site: siteHome');
});

it('renders recipeDetaills correctly', () => {
  const { getAllByTestId } = render(
    <RecipeDetails location={location} />, { wrapper: MemoryRouter },
  );
  const result = getAllByTestId('recipeDetails');
  expect(result[0]).not.toHaveTextContent('any random string');
});

it('matches recipeDetails snapshot', () => {
  const appointment = renderer.create(<Router><RecipeDetails location={location} /></Router>);
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
