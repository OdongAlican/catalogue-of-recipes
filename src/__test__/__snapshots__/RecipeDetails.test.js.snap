// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches recipeDetails snapshot 1`] = `
<div
  data-testid="recipeDetails"
>
  <div
    className="card p-2 col-6 border mx-auto mt-3 border"
  >
    <div
      className="image-section-details mx-auto mb-2 border p-2"
    >
      <img
        alt="boohoo"
        className="image-detail"
        src="sample"
      />
    </div>
    <div
      className="card-body card-section"
    >
      <h5
        className="card-title text-underline"
      >
         
        title
         
      </h5>
      <p
        className="card-text text-secondary"
      />
      <p
        className="card-text text-underline"
      >
         Published Date: 
        Wed Sep 23 2020
         
      </p>
      <p
        className="card-text text-underline"
      >
         Web-site:
        <a
          className="ml-1"
          href="site"
          rel="noopener noreferrer"
          target="_blank"
        >
           
          site
        </a>
      </p>
    </div>
    <button
      className="col-md-2 btn btn-primary"
    >
      <a
        className="text-white"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </button>
  </div>
</div>
`;
